### ISSUE TITLE
Feature Request - Multi Page Layout (Change Orientation/Page Size)

### ISSUE THREAD

**Issue by MotazHakim** [Stirling-Tools/Stirling-PDF#397]
Hello
When opening Multi Page Layout and setting the Pages per sheet to 2
For example, I have a timetable with 2 days and 2 pages, every page has a day, and every page size is A4
now I want to get them together in one page as A4 split to A5 | A5
but it split the A4 page in portrait mode and that's a mistake,
so, this option should also have an Orientation option and the output size 

also, the output size should be added to many options like "PDF to Single Page" I've tested it and it added the 2 pages into 1 page as Portrait , so you should consider add  more controlling options of the output like : control the margins 

and i really wanna thank you alot for the project it helped alot

**Comment by Legendary4226** [comment]
:)

**Comment by Kuchenmampfer** [comment]
If adding more options is not desirable in favor of better defaults, I would
- for one-dimensional output-page-arrays (2 or 3 pages per sheet): Reverse the sheet dimensions and arrange pages so they touch on their long edge
- for two-dimensional output-page-arrays (any square number of pages per sheet): Keep the sheet dimensions and put pages in a grid

In case someone stumbles here and needs another tool for multi page layout as long as stirling pdf is not sufficient, for me, `pdfxup` (which is part of `texlive`) just did the job, see its [man page](https://manpages.ubuntu.com/manpages/oracular/en/man1/pdfxup.1.html).

**Comment by mon-jai** [comment]
It would also be useful if we could customize the grid size. e.g. able to specify a vertical layout with 2 x 8 pages per sheet.

**Comment by balazs-szucs** [comment]
Hi,

Most of the request here can be achieved with Ghostscript;

For vertical/horizontal change you can toggle the /Orientation in the gs script e.g.,:

`gs -sDEVICE=pdfwrite -o output_3x2_landscape.pdf -sNupControl=3x2 -sPAPERSIZE=a4 -dAutoRotatePages=/None -dFIXEDMEDIA -dFitPage -c "<</Orientation 3>> setpagedevice" -f input.pdf`

vs

`gs -sDEVICE=pdfwrite -o output_2x3_portrait.pdf -sNupControl=2x3 -sPAPERSIZE=a4 -dAutoRotatePages=/None -dFIXEDMEDIA -dFitPage -c "<</Orientation 0>> setpagedevice" -f pdf/image-doc.pdf`

Samples I used;

[sample1_2x3.pdf](https://github.com/user-attachments/files/22533206/sample1_2x3.pdf)
[sample2_3x2.pdf](https://github.com/user-attachments/files/22533205/sample2_3x2.pdf)

However generally it's not that simple; column-major order is NOT supported natively by GS so you have to adjust the document beforehand. (this may include; reordering of pages; splitting document into multiple document e.g., you want 4 column but you have 12 pages you have to split the document in to multiple document with 4 pages and merge afterwards)

> The orientation of the sheet

This also can be done; but it has be done before the command is called.

> The direction how pages are placed

Not 100% what's meant here; some stuff we can adjust via implicit paramaters e.g., nested grids, explicit anchor points (dDEVICEWIDTHPOINTS and -dDEVICEHEIGHTPOINTS) or others to achieve a certain look but it should be specified a bit more precisely what you mean here (possibly example doc would be helpful)

> Split (A5 | A5 on A4)

Isn't this the current implementation when you use 2 setting?

For more infos please read: https://ghostscript.readthedocs.io/en/latest/Use.html

Generally, I think this is a good first issue. Not very easy but definitely manageable

**Comment by OUNZAR-Aymane** [comment]
Hi! I will start working on this. I'll see what I can do.  @Frooodle Can you assign me ?

**Comment by xky0007** [comment]
Will this be added in V2? Thanks.

**Comment by OUNZAR-Aymane** [comment]
> Will this be added in V2? Thanks.

I just opened a PR for this.

**Comment by Frooodle** [comment]
It should be out later next week in V2.7.1

**Comment by xky0007** [comment]
> It should be out later next week in V2.7.1

Seems still not available for V2.7.2.

**Comment by Frooodle** [comment]
Yeah sorry it will be out next  major release 2.8.0  we had some issues with other PRs and did some hotfixes so our feature release was delayed
